### ISSUE TITLE
Mobile Ads SDK (iOS) versioning error - Xamarin.Firebase.iOS.Admob plugin

### ISSUE THREAD

**Issue by blmiles** [xamarin/GoogleApisForiOSComponents#520]
Hello,

Methinks someone should update this plugin library as it fails in the app (iOS Xamarin) citing wrong version of the Mobile Ads SDK.

This is what I get -

_2022-01-21 13:56:38.844 MyCompany[3831:1156576] Google:HTML You are currently using version 7.66.0 of the SDK. Please consider updating your SDK to the most recent SDK version (8.13.0) to get the latest features and bug fixes. The latest SDK can be downloaded from https://goo.gl/UoiJ8F. A full list of release notes is available at https://developers.google.com/admob/ios/rel-notes._

Any feedback?

**Comment by dmariogatto** [comment]
@blmiles I've just helped out with keeping the bindings up-to-date. I'm not a project maintainer and I don't have any extra knowledge on the release timeline beyond what is public on the repo.

I personally use AdMob and have been using a custom nuget built from my PR (#501) while I wait for an offical release.

**Comment by blmiles** [comment]
@dmariogatto , thanks for the response Daniel. I've pulled code but there seems to be too many external references to build...

If possible, could you tag whomever **IS** responsible for getting an update out.
There are a lot of people awaiting an update, you and I included, and it's time MS and the Xamarin team get it done.

Thanks

**Comment by blmiles** [comment]
Apart from needing the Xamarin.Firebase.iOS.Admob plugin updated with the latest Google SDK, here's minor work-around in the AdmobViewRenderer as cited in the samples...

When navigating TO various pages, we're not seeing any issues or breaks, however, navigating BACK to a previous page or the main page in a Xamarin.Forms app was causing a crash and the App quits.

This is because the "Element" in the CreateBannerView() method is null and the AdUnitId cannot be set.

I simply set the AdUnitId directly in the CreateBannerView method and this works fine for now.
Obviously I can move the id string into the plist or wherever else you store string vars/consts.

`        private BannerView CreateBannerView()
        {
                //bannerView = new BannerView(AdSizeCons.SmartBannerPortrait)
                //{
                //    AdUnitId = Element.AdUnitId,
                //RootViewController = GetVisibleViewController()
                //};

                bannerView = new BannerView(AdSizeCons.SmartBannerPortrait)
                {
                    AdUnitId = "**ca-app-pub-[your adunitId here]**",
                    RootViewController = GetVisibleViewController()
                };

                bannerView.LoadRequest(GetRequest());

                Request GetRequest()
                {
                    var request = Request.GetDefaultRequest();
                    return request;
                }

                return bannerView;

        }
`

Not tried the Android implementation yet but this makes all GoogleAds work in iOS Xamarin.Forms app.

Simple, but I hope that helps someone...

Thanks

**Comment by Mataboge** [comment]
Finally somebody complained!

**Comment by dmariogatto** [comment]
@blmiles

[AbMob 8.13](https://www.nuget.org/packages/Xamarin.Firebase.iOS.AdMob/) is now on NuGet.

**Comment by tipa** [comment]
The new package is missing the stuff from `Google.MobileAds.Consent`.
Was the ConsentFrom removed from the library or is it just due to missing bindings?
Seems like I am stuck on v7.66 for now.

**Comment by dmariogatto** [comment]
@tipa UMP was removed from the AdMob package and moved to a [separate binding](https://github.com/xamarin/GoogleApisForiOSComponents/tree/main/source/Google/UserMessagingPlatform).

It should've been released, however, looks like the package name is already in [use](https://www.nuget.org/packages/Xamarin.Google.iOS.UserMessagingPlatform) (you might be able to use that package for the time being).

@Redth @SotoiGhost

**Comment by OMendozaG** [comment]
¿ what about Google.MobileAds.Consent ?

**Comment by tipa** [comment]
I am in the process of porting my Xamarin.iOS app to .NET for iOS and I am having problems with this again.
I can use the stuff from the `Google.MobileAds` namespace, but the `Google.UserMessagingPlatform` namespace is not available, event though I referenced the `UserMessagingPlatform` Nuget. Perhaps that package hasn't been built for the `net6.0-ios` target yet as the `MobileAds` package has been?

**Comment by tipa** [comment]
@Redth are the UMP bindings still supported and if so, can we expect working bindings and a new Nuget version at some point or should we expect this binding to be deprecated and not supported by Microsoft?

**Comment by pboisso** [comment]
@tipa I'm interested to know if you've been able to successfully have UMP working on Xamarin.iOS?

**Comment by tipa** [comment]
@pboisso Yes I have, see [here](https://github.com/xamarin/GoogleApisForiOSComponents/issues/628#issuecomment-1792011764)
